packages/eas-cli/src/channel/queries.ts

```diff
@@ -264,3 +268,46 @@ export async function ensureChannelExistsAsync(
     }
   }
 }
+
+export async function getChannelsDatasetAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    appId,
+    filterPredicate,
+    batchSize = 100,
+  }: {
+    appId: string;
+    filterPredicate?: (channelInfo: UpdateChannelBasicInfoFragment) => boolean;
+    batchSize?: number;
+  }
+): Promise<UpdateChannelBasicInfoFragment[]> {
+  const queryAsync = async ({
+    first,
+    after,
+  }: QueryParams): Promise<Connection<UpdateChannelBasicInfoFragment>> =>
+    await ChannelQuery.viewUpdateChannelsBasicInfoPaginatedOnAppAsync(graphqlClient, {
+      appId,
+      first,
+      after,
+    });
+
+  const assetSpinner = ora().start('Fetching channels...');
+  const afterEachQuery = (
+    totalNodesFetched: number,
+    _dataset: UpdateChannelBasicInfoFragment[],
+    _batch: UpdateChannelBasicInfoFragment[],
+    pageInfo: PageInfo
+  ): void => {
+    if (pageInfo.hasNextPage) {
+      assetSpinner.text = `Fetched ${totalNodesFetched} channels`;
+    }
+  };
+  const dataset = await getPaginatedDatasetAsync({
+    queryAsync,
+    afterEachQuery,
+    filterPredicate,
+    batchSize,
+  });
+  assetSpinner.succeed(`Fetched all channels`);
```

Comment:
this message could be incorrect if `maxNodesFetched` was less than the total, but I think it likely isn't an issue